app/models/push_notification_target.rb

```diff
@@ -0,0 +1,113 @@
+class PushNotificationTarget < ApplicationRecord
+  resourcify
+
+  FOREM_BUNDLE = "com.forem.app".freeze
+  SUPPORTED_PLATFORMS = [Device::ANDROID, Device::IOS].freeze
+  FOREM_APP_PLATFORMS = [Device::IOS].freeze
+
+  validates :app_bundle, presence: true
+  validates :platform, presence: true
+
+  # Clear Redis-backed model to ensure it will be recreated with updated values
+  after_update :clear_rpush_app
+
+  def forem_app?
+    app_bundle == FOREM_BUNDLE
+  end
+
+  def active?
+    # TRUE if it's marked as `active` in the DB && it has credentials available
+    active && auth_credentials?
+  end
```

Comment:
When sending a Push Notification is attempted but the delivery fails because the credentials expired or aren't valid (see callbacks from [Reflection API](https://github.com/rpush/rpush/wiki/Reflection-API)) we can mark the `PushNotificationTarget` as invalid. This will provide visual feedback and will be easier to identify when PN delivery credentials aren't working and need to be re-issued.